src/useradd.c

```diff
@@ -1964,16 +1964,26 @@ static void faillog_reset (uid_t uid)
 	memzero (&fl, sizeof (fl));
 
 	fd = open (FAILLOG_FILE, O_RDWR);
-	if (   (-1 == fd)
-	    || (lseek (fd, offset_uid, SEEK_SET) != offset_uid)
+	if (-1 == fd) {
+		fprintf (stderr,
+		         _("%s: failed to open the faillog file for UID %lu: %s\n"),
+		         Prog, (unsigned long) uid, strerror (errno));
+		SYSLOG ((LOG_WARN, "failed to open the faillog file for UID %lu", (unsigned long) uid));
+		return;
+	}
+	if (   (lseek (fd, offset_uid, SEEK_SET) != offset_uid)
 	    || (write (fd, &fl, sizeof (fl)) != (ssize_t) sizeof (fl))
-	    || (fsync (fd) != 0)
-	    || (close (fd) != 0)) {
+	    || (fsync (fd) != 0)) {
 		fprintf (stderr,
 		         _("%s: failed to reset the faillog entry of UID %lu: %s\n"),
 		         Prog, (unsigned long) uid, strerror (errno));
 		SYSLOG ((LOG_WARN, "failed to reset the faillog entry of UID %lu", (unsigned long) uid));
-		/* continue */
```

Comment:
Sorry for the nit (of sorts), but originally all these cases continued on error.  In the first case you are now return()ing, and in the others you are not returning but dropped the comment making clear that continuing was a conscious decision.

What do you think, should we keep the comment and continue in all cases?  Change the behavior?

Reply:
IMO, we should return if the file is not opened as I don't think it makes any sense to operate on a file that hasn't been opened. In any other circumstance we should continue and I've updated the code accordingly to include the comment again.

Reply:
Huh. I was thinking this was happening in a loop.  Right you are.  Thanks.